### PR TITLE
LPS-162323 | Add PagesPropagatedFromSiteTemplateCanReferenceMasterPage testcase

### PR DESCRIPTION
**Pull request by alessandromalves**
https://issues.liferay.com/browse/LPS-162323

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by alessandromalves** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/macros/site/SiteTemplates.macro [hunk 1]
```diff
@@ -84,6 +84,9 @@ definition {
 				locator1 = "ContentRow#ENTRY_CONTENT_ENTRY_NAME",
 				value1 = "${rowEntry}");
 		}
+		else if (isSet(masterLayoutName)) {
+			LexiconCard.clickCard(card = "${masterLayoutName}");
+		}
 		else if ("${rowEntry}" == "Content Page") {
 			LexiconCard.clickCard(card = "Blank");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/site/SiteTemplates.macro:91]: Replace line 91 by
```
		ContentPages._selectCard(masterLayoutName = "${masterLayoutName}");
```

File: portal-web/test/functional/com/liferay/portalweb/macros/site/SiteTemplates.macro [hunk 2]
```diff
@@ -84,6 +84,9 @@ definition {
 				locator1 = "ContentRow#ENTRY_CONTENT_ENTRY_NAME",
 				value1 = "${rowEntry}");
 		}
+		else if (isSet(masterLayoutName)) {
+			LexiconCard.clickCard(card = "${masterLayoutName}");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/site/SiteTemplates.macro:89]: Remove this change

File: portal-web/test/functional/com/liferay/portalweb/macros/site/SiteTemplates.macro [hunk 3]
```diff
@@ -107,7 +110,7 @@ definition {
 			PagesAdmin.togglePageTemplateInheritChanges(toggleValue = "${inheritChanges}");
 		}
 
-		if ("${sitePageType}" != "Content Page") {
+		if (IsElementNotPresent(locator1 = "PageEditor#FRAGMENT_SIDEBAR_BODY")) {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/site/SiteTemplates.macro:113]: Keep the original

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/exportimport/ExportImport.testcase [hunk 4]
```diff
@@ -544,7 +544,7 @@ definition {
 
 			SiteTemplates.addSitePageViaSiteTemplateCP(
 				pageName = "Test Template Content Page",
-				sitePageType = "Content Page",
+				rowEntry = "Content Page",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/exportimport/ExportImport.testcase:547]: Keep the original

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 5]
```diff
@@ -1132,6 +1132,89 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-162323. The Pages propagated from a Site Template should reference the Master Page of the Site Template."
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1135]: The pages propagated from a Site Template should refer to the master page from site

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 6]
```diff
@@ -1132,6 +1132,89 @@ definition {
...
+	@priority = "4"
+	test PagesPropagatedFromSiteTemplateCanReferenceMasterPage {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1137]: ViewReferredMasterPageOfPagesPropagatedFromSiteTemplate

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 7]
```diff
@@ -1132,6 +1132,89 @@ definition {
...
+		task ("Given a page creator has a site with a master page from a site template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1138]: Given a page creator has a site with a page from a site template

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 8]
```diff
@@ -1132,6 +1132,89 @@ definition {
...
+				ProductMenu.gotoPortlet(
+					category = "Design",
+					portlet = "Page Templates");
+
+				MastersPageTemplatesAdmin.addMasters(masterLayoutName = "Custom Masters Page Template");
+
+				MastersPageTemplates.publishMaster();
+			}
+
+			task ("Add a page into the site template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1161]: Add a page based on master page in the site template

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 9]
```diff
@@ -1132,6 +1132,89 @@ definition {
...
+
+			task ("Add a site based on site template") {
+				Site.openSitesAdmin();
+
+				Site.addSiteTemplateCP(
+					siteName = "Test Site Name 2",
+					siteTemplateName = "Site Template Name");
+			}
+
+			task ("Add another page into the site template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1178]: Add another page based on master page in the site template

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 10]
```diff
@@ -1132,6 +1132,89 @@ definition {
...
+				SiteTemplates.addSitePageViaSiteTemplateCP(
+					masterLayoutName = "Custom Masters Page Template",
+					pageName = "Test Page Name 2",
+					siteTemplateName = "Site Template Name");
+
+				PageEditor.publish();
+			}
+		}
+
+		task ("When the page creator changes the master template page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1188]: When the page creator makes some changes in master page from site

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 11]
```diff
@@ -1132,6 +1132,89 @@ definition {
...
+			PageEditor.addFragment(
+				collectionName = "Basic Components",
+				dropTarget = "Header",
+				fragmentName = "Heading",
+				scope = "Master");
+
+			MastersPageTemplates.propagateChange();
+		}
+
+		task ("Then the page creator modifications can be seen on both pages at view mode") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1204]: Then the page creator can see changes on propagated pages

**Review by Tim-Cao — COMMENTED** [review]
Hi @alessandromalves, would you mind updating according to comments? Thanks

**Comment by alessandromalves** [comment]
ci:test:sf

**Comment by alessandromalves** [comment]
> Hi @alessandromalves, would you mind updating according to comments? Thanks

Updated! thank you!

**Comment by Tim-Cao** [comment]
Good job! @alessandromalves , LGTM!

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**